### ISSUE TITLE
Please add aria-label with cansel-button-text and reset-button-text ion-searcbar

### ISSUE THREAD

**Issue by terrancle** [ionic-team/ionic#21013]
Hello,
With blind users, the cansel-button-text is not configured for aria-label.
Identical problem with reset button.
In addition, the cansel-button-text is displayed all the time for screen readers like VoiceOver, even if the keyboard is not open.

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you please describe the issue you are running into?

**Comment by terrancle** [comment]
When the cancelButtonText attribute is set, the aria-label attribute should be set to the same value.
Currently VoiceOver sees the value of the default button "Cancel", while cancelButtonText has another value.
This is mainly a problem in the case of a translation in several languages, when cancelButtonText contains the value of the language of the user.
If I write in French "Annuler" for the cancel translation, VoiceOver will say "Cancel", because the aria-label attribute is not present in the ionButton of ionSearcBar.

**Comment by terrancle** [comment]
Identical problem for the clearButtonText, and ionIcon "Search outline".
Please add aria-hidden="true" for the "Search outline" icon.

**Comment by liamdebeasi** [comment]
Thanks for the follow up. Can you try the following dev build and let me know if it resolves the issue?

Ionic Angular
`npm i @ionic/angular@5.1.0-dev.202004152018.e90683a`

Ionic React
`npm i @ionic/react@5.1.0-dev.202004152018.e90683a`

**Comment by terrancle** [comment]
Thank you very much for your response,
Can you also add an aria-hidden = "true" to the searchIcon icon?
I will perform the other tests.

**Comment by liamdebeasi** [comment]
All of the `ion-icon` elements in the searchbar have `aria-hidden="true"` already. What issue are you running into with them?

**Comment by terrancle** [comment]
Okay, that's not the case in production version 5.0.7.
VoiceOver states:
* The search field,
* The "Search outline" icon for example,
* Cancel button
I will see during my test on version 5.1.0DEV, maybe this was corrected by another request.
At the same time, I detected a problem.
When showCancelButton = "focus", VoiceOver sees the CancelButtonText button all the time, even when the keyboard is not open.
Is this problem resolved in 5.1.0DEV?
If you run VoiceOver without opening the search, and if cancelButtonText is set, does VoiceOver see the button?
I'm not sure how to advise you on how to fix this problem. Perhaps you should put an aria-hidden = "true" on the Cancel button, and redisplay it with an aria-live when the search keyboard is open, if showCancelButton = "focus".
I hope I can explain myself well.
Big thanks to you.

**Comment by liamdebeasi** [comment]
Yeah, I was referring to the `aria-hidden="true"` being applied in the dev version. Is it working properly for you in the dev version?

**Comment by liamdebeasi** [comment]
Here is another dev build. In addition to the original fixes, VoiceOver should no longer announce the cancel button when the cancel button is not shown on the screen.

Ionic Angular
`npm i @ionic/angular@5.1.0-dev.202004161533.0cd46d9`

Ionic React
`npm i @ionic/react@5.1.0-dev.202004161533.0cd46d9`

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic/pull/21049, and a fix will be available in an upcoming release of Ionic Framework.
